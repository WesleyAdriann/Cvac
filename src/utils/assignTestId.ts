import { Platform } from 'react-native'

interface ITestID {
  accessible?: boolean
  testID: string
  accessibilityLabel?: string
}

type TComponentType =
  | 'View'
  | 'ScrollView'
  | 'TextInput'
  | 'Text'
  | 'SafeAreaView'
  | 'Pressable'
  | 'Modal'

export const assignTestId = (
  componentType: TComponentType,
  componentId: string
): ITestID => {
  const id = `${componentType}_${componentId}`
  return Platform.OS === 'android'
    ? {
        testID: id,
        accessible: true,
        accessibilityLabel: id
      }
    : {
        testID: id,
        accessible: false
      }
}
export const mockTestID = (
  componentType: TComponentType,
  componentId: string
): ITestID => assignTestId(componentType, componentId)
