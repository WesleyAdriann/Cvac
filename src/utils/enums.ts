export enum ECalendarsName {
  'kid' = 'Crian√ßa',
  'elder' = 'Idoso',
  'adult' = 'Adulto',
  'teen' = 'Adolescente'
}
