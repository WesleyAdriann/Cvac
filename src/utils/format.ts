export const formatDate = (value: string) => {
  const clearDate = value.replace(/\D/g, '')

  if (clearDate.length === 8) return value

  const day = clearDate.slice(0, 2)
  const month = clearDate.slice(2, 4)
  const year = clearDate.slice(4, 8)

  if (!month) return day
  if (!year) return `${day}/${month}`
  return `${day}/${month}/${year}`
}
