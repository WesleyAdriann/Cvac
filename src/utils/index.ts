export * from './assignTestId'
export * from './withProviders'
export * from './format'
export * from './logs'
