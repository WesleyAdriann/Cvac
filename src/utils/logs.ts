export const log = (TAG: string, ...args: any[]) => {
  if (__DEV__) console.log(`[${TAG}] `, ...args)
}
