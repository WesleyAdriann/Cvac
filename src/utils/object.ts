export const isEmpty = (obj: object | null) => {
  if (!obj) return true
  // eslint-disable-next-line no-unreachable-loop
  for (const key in obj) return false
  return true
}
