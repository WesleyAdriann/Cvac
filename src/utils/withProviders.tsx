import React from 'react'
import { Provider as PaperProvider } from 'react-native-paper'
import { ThemeProvider } from 'styled-components'

import { theme, paperTheme } from '../tokens'

export const TestsProviders: React.FC = ({ children }) => (
  <PaperProvider theme={paperTheme} >
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </PaperProvider>
)
