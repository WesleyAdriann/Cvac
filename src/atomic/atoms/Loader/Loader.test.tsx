import React from 'react'
import { render, cleanup } from '@testing-library/react-native'

import { AllTheProviders as wrapper, mockTestID } from '../../../utils'

import { Loader, ILoader } from './index'

describe('Atoms/Loader', () => {
  const props: ILoader = {
    testID: 'Loader'
  }

  const elContainer = mockTestID('View', props.testID!).testID

  const renderComponent = () => render(<Loader {...props} />, { wrapper })
  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  it(`should render component #${elContainer}`, async () => {
    const sut = await component.findByTestId(elContainer)
    expect(sut).toBeTruthy()
  })

  it('should render snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
