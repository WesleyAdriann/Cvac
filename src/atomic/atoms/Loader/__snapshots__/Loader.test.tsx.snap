// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Loader should render snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityRole="progressbar"
    accessibilityState={
      Object {
        "busy": true,
      }
    }
    accessible={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Array [
          Object {
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
          },
        ],
      ]
    }
    testID="View_Loader"
  >
    <View
      collapsable={false}
      style={
        Object {
          "height": 24,
          "opacity": 1,
          "width": 24,
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "height": 24,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 12,
                "overflow": "hidden",
                "width": 24,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "rotate": "-165deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  style={
                    Object {
                      "borderColor": "#00C9D2",
                      "borderRadius": 12,
                      "borderWidth": 2.4,
                      "height": 24,
                      "width": 24,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "height": 24,
              "transform": Array [
                Object {
                  "rotate": "45deg",
                },
              ],
              "width": 24,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "height": 12,
                "overflow": "hidden",
                "top": 12,
                "width": 24,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 24,
                  "transform": Array [
                    Object {
                      "translateY": -12,
                    },
                    Object {
                      "rotate": "345deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  style={
                    Object {
                      "borderColor": "#00C9D2",
                      "borderRadius": 12,
                      "borderWidth": 2.4,
                      "height": 24,
                      "width": 24,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
